ga/src/main/java/nl/tudelft/serg/evosql/metaheuristics/Approach.java

```diff
@@ -68,35 +75,50 @@ protected Fixture minimize (Fixture fixture) throws SQLException {
 					minimizedFixture.remove(ft.getName(), i);
 					i--;
 				}
-			}	
+			}
 		}
-		
+
 		return minimizedFixture;
 	}
-	
-	public boolean hasOutput(Fixture fixture) throws SQLException {
-		return hasOutput(fixture, "", -1);
+
+	public DatabaseOutput generateOutput(Fixture fixture, String sql) throws SQLException {
```

Comment:
(super nitpick) Technically it's not really generating data but merely fetching it. It's not a blocking issue in my opinion.

Reply:
I can refactor this with the other changes